osquery/tables/system/darwin/firewall.mm

```diff
@@ -67,12 +70,98 @@ QueryData parseALFTree(const pt::ptree& tree) {
 }
 
 QueryData genALF(QueryContext& context) {
-  pt::ptree tree;
-  auto s = genALFTreeFromFilesystem(tree);
-  if (!s.ok()) {
+  const auto& qd = SQL::selectAllFrom("os_version");
```

Comment:
My memory is that there's some effort to avoid calling SQL select from within tables -- it's part of some long range issue around test size, and include files. But since I know we do it a bunch, and I don't offhand know the alternative, I'm not blocking over it

Reply:
The better alternative is to export the common logic needed from the tables in utility functions, instead of using SQL and the tables.
In some cases it's a matter of performance, in other it's a matter of unnecessary (or even problematic) dependencies among tables or other parts of codes.

The worse offender is the core depending on tables. It's a dependency loop and creates that necessity we have to include all symbols from archive files, even if actually unused, which in turn causes test binaries to be unnecessarily big, slowing down compilation and hitting disk limits.

Reply:
Sounds like some overall effort is needed to refactor all of them. Otherwise people like me will keep mimicking existing code :)

Reply:
Yeah it's a process that I started multiple times, arriving also at seeing the improvements in size (3x reduction was the minimum), but time is an issue when doing it all together.

That being said, converting tables one by one can still be done, and then a policy to avoid getting more.
It's mostly a matter on the reviewers being on the same page on what's the direction.